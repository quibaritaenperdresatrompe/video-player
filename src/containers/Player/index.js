import glamorous from 'glamorous'
import PropTypes from 'prop-types'
import React, { Component } from 'react'

import ControlsBar from '../../components/ControlsBar'
import RangeBar from '../../components/RangeBar'

export const SPACE_KEY_CODE = 32
export const ESC_KEY_CODE = 27

const PlayerContainer = glamorous.div(
  ({ isCursorHidden }) => ({
    cursor: isCursorHidden ? 'none' : 'default',
  }),
  ({ isFullscreenMode }) => {
    if (isFullscreenMode) {
      return {
        backgroundColor: 'hsl(0, 0%, 16%)',
        bottom: 0,
        height: '100vh',
        left: 0,
        margin: 'auto',
        position: 'absolute',
        width: '100vw',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
      }
    }
    return {
      position: 'relative',
    }
  },
)

const ControlsContainer = glamorous.div({
  background: 'linear-gradient(bottom, hsl(0, 0%, 14%) 0%, transparent 100%)',
  bottom: 0,
  left: 0,
  position: 'absolute',
  width: 'calc(100% - 1em)',
  padding: '0.5em',
})

class Player extends Component {
  constructor() {
    super()
    this.state = {
      autoHideControlsBarTimeout: null,
      currentTime: 0,
      duration: 0,
      isComplete: false,
      isControlsBarHidden: false,
      isFocused: false,
      isFullscreenMode: false,
      isMuted: false,
      isPlaying: false,
      volume: 1,
    }
  }

  componentWillMount() {
    window.addEventListener('keyup', this.handleKeyUp)
  }

  componentWillUnmount() {
    window.removeEventListener('keyup', this.handleKeyUp)
  }

  handleEnterFullscreenMode = () => {
    this.setState(() => ({
      isFullscreenMode: true,
    }))
  }

  handleExitFullscreenMode = () => {
    this.setState(() => ({
      isFullscreenMode: false,
    }))
  }

  handleKeyUp = ({ keyCode }) => {
    if (this.state.isFocused) {
      switch (keyCode) {
        case SPACE_KEY_CODE:
          if (this.state.isPlaying) this.handlePause()
          else this.handlePlay()
          break
        case ESC_KEY_CODE:
          if (this.state.isFullscreenMode) this.handleExitFullscreenMode()
          break
        default:
          break
      }
    }
  }

  handleLoad = () => {
    this.setState(() => ({
      duration: this.player.duration,
    }))
  }

  handleMouseEnter = () => {
    this.showControlsBar(true)
    this.setState(() => ({
      isFocused: true,
    }))
  }

  handleMouseLeave = () => {
    this.hideControlsBar()
    this.setState(() => ({
      isFocused: false,
    }))
  }

  handleMouseMove = () => {
    this.showControlsBar(true)
  }

  handleMute = () => {
    this.mute()
    this.setState(() => ({
      isMuted: true,
    }))
  }

  handlePause = () => {
    this.pause()
    this.setState(() => ({
      isPlaying: false,
    }))
    this.showControlsBar()
  }

  handlePlay = () => {
    this.play()
    this.setState(() => ({
      isPlaying: true,
    }))
    this.hideControlsBar(true)
  }

  handleSeekTo = currentTime => {
    this.player.currentTime = currentTime
    this.setState(() => ({ currentTime }))
  }

  handleTimeUpdate = () => {
    this.setState(() => ({
      currentTime: this.player.currentTime,
      isComplete: this.player.currentTime === this.state.duration,
    }))
  }

  handleUnmute = () => {
    this.unmute()
    this.setState(() => ({
      isMuted: false,
    }))
  }

  handleUpdateVolumeTo = volume => {
    this.player.volume = volume
    this.setState(() => ({ volume }))
  }

  hideControlsBar = (isDelayed = false) => {
    if (isDelayed) {
      this.hideControlsBarWithDelay()
    } else {
      this.setState(() => ({
        isControlsBarHidden: true,
      }))
    }
  }

  hideControlsBarWithDelay = () => {
    const timeout = window.setTimeout(this.hideControlsBar, 3e3)
    this.setState(() => ({
      autoHideControlsBarTimeout: timeout,
    }))
  }

  mute = () => {
    this.player.muted = true
  }

  unmute = () => {
    this.player.muted = false
  }

  play = () => this.player.play()

  pause = () => this.player.pause()

  showControlsBar = (isTemporary = false) => {
    window.clearTimeout(this.state.autoHideControlsBarTimeout)
    this.setState(
      () => ({
        isControlsBarHidden: false,
      }),
      () => {
        if (isTemporary) this.hideControlsBarWithDelay()
      },
    )
  }

  renderControlsBar = () => {
    const {
      currentTime,
      duration,
      isComplete,
      isControlsBarHidden,
      isFullscreenMode,
      isMuted,
      isPlaying,
      volume,
    } = this.state

    if (isPlaying && !isComplete && isControlsBarHidden) return null
    return (
      <ControlsContainer className='Controls'>
        <RangeBar
          currentValue={currentTime}
          maxValue={duration}
          setTo={this.handleSeekTo}
          color='hsl(0, 68%, 50%)'
        />
        <ControlsBar
          currentTime={currentTime}
          duration={duration}
          enterFullscreenMode={this.handleEnterFullscreenMode}
          exitFullscreenMode={this.handleExitFullscreenMode}
          isComplete={isComplete}
          isFullscreenMode={isFullscreenMode}
          isMuted={isMuted}
          isPlaying={isPlaying}
          mute={this.handleMute}
          pause={this.handlePause}
          play={this.handlePlay}
          unmute={this.handleUnmute}
          updateVolumeTo={this.handleUpdateVolumeTo}
          volume={volume}
        />
      </ControlsContainer>
    )
  }

  render() {
    const {
      isControlsBarHidden,
      isComplete,
      isFullscreenMode,
      isPlaying,
    } = this.state
    const controlAction = isPlaying && !isComplete ? this.handlePause : this.handlePlay
    const isCursorHidden = isPlaying && isControlsBarHidden

    if (!this.props.medium) return null
    return (
      <div>
        <PlayerContainer
          isCursorHidden={isCursorHidden}
          isFullscreenMode={isFullscreenMode}
          onMouseEnter={this.handleMouseEnter}
          onMouseLeave={this.handleMouseLeave}
          onMouseMove={this.handleMouseMove}
        >
          <video
            height={isFullscreenMode ? '100%' : this.props.height}
            onClick={controlAction}
            onLoadedMetadata={this.handleLoad}
            onTimeUpdate={this.handleTimeUpdate}
            ref={el => { this.player = el }}
            src={this.props.medium.source}
          />
          {this.renderControlsBar()}
        </PlayerContainer>
        <h3>{this.props.medium.title}</h3>
      </div>
    )
  }
}

Player.propTypes = {
  medium: PropTypes.shape({
    title: PropTypes.string,
    source: PropTypes.string,
  }),
  height: PropTypes.number,
}

Player.defaultProps = {
  medium: null,
  height: 480,
}

export default Player
