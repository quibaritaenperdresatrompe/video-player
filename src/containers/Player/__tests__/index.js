import Player, { SPACE_KEY_CODE, ESC_KEY_CODE } from '../'

import itRendersAllMutations from '../../../utils/jest-it-renders-all-mutations'

const mutations = [
  {
    name: 'without props',
    props: null,
  },
  {
    name: 'with medium',
    props: {
      medium: {},
    },
  },
  {
    name: 'when ControlsBar is hidden',
    props: {
      medium: {},
    },
    state: {
      isComplete: false,
      isControlsBarHidden: true,
      isPlaying: true,
    },
  },
  {
    name: 'when is playing',
    props: {
      medium: {},
    },
    state: {
      isPlaying: true,
    },
  },
  {
    name: 'when is in fullscreen mode',
    props: {
      medium: {},
    },
    state: {
      isFullscreenMode: true,
    },
  },
]

describe(Player.name, () => {
  itRendersAllMutations(Player, mutations)

  describe('lifecycle', function scope() {
    beforeEach(() => {
      this.Player = new Player()
    })

    describe('componentWillMount', () => {
      test('it calls addEventListener', () => {
        this.Player.handleKeyUp = jest.fn()
        window.addEventListener = jest.fn()
        this.Player.componentWillMount()
        expect(window.addEventListener).toHaveBeenCalledWith('keyup', this.Player.handleKeyUp)
      })
    })

    describe('componentWillUnmount', () => {
      test('it calls removeEventListener', () => {
        this.Player.handleKeyUp = jest.fn()
        window.removeEventListener = jest.fn()
        this.Player.componentWillUnmount()
        expect(window.removeEventListener).toHaveBeenCalledWith('keyup', this.Player.handleKeyUp)
      })
    })
  })

  describe('handlers', function scope() {
    beforeEach(() => {
      this.Player = new Player()
    })

    describe('handleEnterFullscreenMode', () => {
      test('it calls setState', () => {
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleEnterFullscreenMode()
      })
    })

    describe('handleExitFullscreenMode', () => {
      test('it calls setState', () => {
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleExitFullscreenMode()
      })
    })

    describe('handleKeyUp', () => {
      test('it does nothing', () => {
        this.Player.state = {
          isFocused: false,
        }
        this.Player.handlePause = jest.fn()
        this.Player.handlePlay = jest.fn()
        this.Player.handleExitFullscreenMode = jest.fn()
        this.Player.handleKeyUp({})
        expect(this.Player.handlePause).not.toHaveBeenCalled()
        expect(this.Player.handlePlay).not.toHaveBeenCalled()
        expect(this.Player.handleExitFullscreenMode).not.toHaveBeenCalled()
      })
      test('it does nothing', () => {
        this.Player.state = {
          isFocused: true,
        }
        const event = { keyCode: 0 }
        this.Player.handlePause = jest.fn()
        this.Player.handlePlay = jest.fn()
        this.Player.handleExitFullscreenMode = jest.fn()
        this.Player.handleKeyUp(event)
        expect(this.Player.handlePause).not.toHaveBeenCalled()
        expect(this.Player.handlePlay).not.toHaveBeenCalled()
        expect(this.Player.handleExitFullscreenMode).not.toHaveBeenCalled()
      })
      test('it calls handlePause', () => {
        this.Player.state = {
          isFocused: true,
          isPlaying: true,
        }
        const event = { keyCode: SPACE_KEY_CODE }
        this.Player.handlePause = jest.fn()
        this.Player.handlePlay = jest.fn()
        this.Player.handleKeyUp(event)
        expect(this.Player.handlePause).toHaveBeenCalled()
        expect(this.Player.handlePlay).not.toHaveBeenCalled()
      })
      test('it calls handlePlay', () => {
        this.Player.state = {
          isFocused: true,
          isPlaying: false,
        }
        const event = { keyCode: SPACE_KEY_CODE }
        this.Player.handlePause = jest.fn()
        this.Player.handlePlay = jest.fn()
        this.Player.handleKeyUp(event)
        expect(this.Player.handlePause).not.toHaveBeenCalled()
        expect(this.Player.handlePlay).toHaveBeenCalled()
      })
      test('it does nothing', () => {
        this.Player.state = {
          isFocused: true,
          isFullscreenMode: false,
        }
        const event = { keyCode: ESC_KEY_CODE }
        this.Player.handleExitFullscreenMode = jest.fn()
        this.Player.handleKeyUp(event)
        expect(this.Player.handleExitFullscreenMode).not.toHaveBeenCalled()
      })
      test('it calls handleExitFullscreenMode', () => {
        this.Player.state = {
          isFocused: true,
          isFullscreenMode: true,
        }
        const event = { keyCode: ESC_KEY_CODE }
        this.Player.handleExitFullscreenMode = jest.fn()
        this.Player.handleKeyUp(event)
        expect(this.Player.handleExitFullscreenMode).toHaveBeenCalled()
      })
    })

    describe('handleLoad', () => {
      test('it calls setState', () => {
        this.Player.player = {
          duration: 99,
        }
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleLoad()
      })
    })

    describe('handleMouseEnter', () => {
      test('it calls showControlsBar and setState', () => {
        this.Player.showControlsBar = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleMouseEnter()
        expect(this.Player.showControlsBar).toHaveBeenCalledWith(true)
      })
    })

    describe('handleMouseLeave', () => {
      test('it calls hideControlsBar and setState', () => {
        this.Player.hideControlsBar = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleMouseLeave()
        expect(this.Player.hideControlsBar).toHaveBeenCalled()
      })
    })

    describe('handleMouseMove', () => {
      test('it calls showControlsBar', () => {
        this.Player.showControlsBar = jest.fn()
        this.Player.handleMouseMove()
        expect(this.Player.showControlsBar).toHaveBeenCalledWith(true)
      })
    })

    describe('handleMute', () => {
      test('it calls mute and setState', () => {
        this.Player.mute = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleMute()
        expect(this.Player.mute).toHaveBeenCalled()
      })
    })

    describe('handlePause', () => {
      test('it calls pause, showControlsBar, and setState', () => {
        this.Player.pause = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.showControlsBar = jest.fn()
        this.Player.handlePause()
        expect(this.Player.pause).toHaveBeenCalled()
        expect(this.Player.showControlsBar).toHaveBeenCalled()
      })
    })

    describe('handlePlay', () => {
      test('it calls play, hideControlsBar, and setState', () => {
        this.Player.play = jest.fn()
        this.Player.hideControlsBar = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handlePlay()
        expect(this.Player.play).toHaveBeenCalled()
        expect(this.Player.hideControlsBar).toHaveBeenCalledWith(true)
      })
    })

    describe('handleSeekTo', () => {
      test('it calls setState and sets player.currentTime', () => {
        this.Player.player = {
          currentTime: 9,
        }
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleSeekTo(10)
        expect(this.Player.player.currentTime).toBe(10)
      })
    })

    describe('handleTimeUpdate', () => {
      test('it calls setState', () => {
        this.Player.state = {
          duration: 99,
        }
        this.Player.player = {
          currentTime: 9,
        }
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleTimeUpdate()
      })
    })

    describe('handleUpdateVolumeTo', () => {
      test('it calls setState and sets player.volume', () => {
        this.Player.player = {
          volume: 0.9,
        }
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleUpdateVolumeTo(1)
        expect(this.Player.player.volume).toBe(1)
      })
    })

    describe('handleUnmute', () => {
      test('it calls unmute and setState', () => {
        this.Player.unmute = jest.fn()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.handleUnmute()
        expect(this.Player.unmute).toHaveBeenCalled()
      })
    })
  })

  describe('functions', function scope() {
    beforeEach(() => {
      this.Player = new Player()
    })

    describe('hideControlsBar', () => {
      test('it calls hideControlsBarWithDelay', () => {
        this.Player.hideControlsBarWithDelay = jest.fn()
        this.Player.hideControlsBar(true)
        expect(this.Player.hideControlsBarWithDelay).toHaveBeenCalled()
      })
      test('it calls setState', () => {
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.hideControlsBar()
      })
    })

    describe('hideControlsBarWithDelay', () => {
      test('it calls window.setTimeout and setState', () => {
        jest.useFakeTimers()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.hideControlsBarWithDelay()
        expect(window.setTimeout).toHaveBeenCalled()
      })
    })

    describe('mute', () => {
      test('it sets player.muted to true', () => {
        this.Player.player = {
          muted: false,
        }
        this.Player.mute()
        expect(this.Player.player.muted).toBe(true)
      })
    })

    describe('unmute', () => {
      test('it sets player.muted to false', () => {
        this.Player.player = {
          muted: true,
        }
        this.Player.unmute()
        expect(this.Player.player.muted).toBe(false)
      })
    })

    describe('play', () => {
      test('it calls player.play', () => {
        this.Player.player = {
          play: jest.fn(),
        }
        this.Player.play()
        expect(this.Player.player.play).toHaveBeenCalled()
      })
    })

    describe('pause', () => {
      test('it calls player.pause', () => {
        this.Player.player = {
          pause: jest.fn(),
        }
        this.Player.pause()
        expect(this.Player.player.pause).toHaveBeenCalled()
      })
    })

    describe('showControlsBar', () => {
      test('it calls window.clearTimeout, setState', () => {
        this.Player.hideControlsBarWithDelay = jest.fn()
        jest.useFakeTimers()
        this.Player.setState = jest.fn((fn) => {
          expect(fn(this.Player.state)).toMatchSnapshot()
        })
        this.Player.showControlsBar()
        expect(window.clearTimeout).toHaveBeenCalled()
      })
    })
  })
})
