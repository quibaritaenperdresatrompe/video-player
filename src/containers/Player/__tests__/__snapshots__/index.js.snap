// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player functions hideControlsBar it calls setState 1`] = `
Object {
  "isControlsBarHidden": true,
}
`;

exports[`Player functions hideControlsBarWithDelay it calls window.setTimeout and setState 1`] = `
Object {
  "autoHideControlsBarTimeout": 1,
}
`;

exports[`Player functions showControlsBar it calls window.clearTimeout, setState 1`] = `
Object {
  "isControlsBarHidden": false,
}
`;

exports[`Player handlers handleEnterFullscreenMode it calls setState 1`] = `
Object {
  "isFullscreenMode": true,
}
`;

exports[`Player handlers handleExitFullscreenMode it calls setState 1`] = `
Object {
  "isFullscreenMode": false,
}
`;

exports[`Player handlers handleLoad it calls setState 1`] = `
Object {
  "duration": 99,
}
`;

exports[`Player handlers handleMouseEnter it calls showControlsBar and setState 1`] = `
Object {
  "isFocused": true,
}
`;

exports[`Player handlers handleMouseLeave it calls hideControlsBar and setState 1`] = `
Object {
  "isFocused": false,
}
`;

exports[`Player handlers handleMute it calls mute and setState 1`] = `
Object {
  "isMuted": true,
}
`;

exports[`Player handlers handlePause it calls pause, showControlsBar, and setState 1`] = `
Object {
  "isPlaying": false,
}
`;

exports[`Player handlers handlePlay it calls play, hideControlsBar, and setState 1`] = `
Object {
  "isPlaying": true,
}
`;

exports[`Player handlers handleSeekTo it calls setState and sets player.currentTime 1`] = `
Object {
  "currentTime": 10,
}
`;

exports[`Player handlers handleTimeUpdate it calls setState 1`] = `
Object {
  "currentTime": 9,
  "isComplete": false,
}
`;

exports[`Player handlers handleUnmute it calls unmute and setState 1`] = `
Object {
  "isMuted": false,
}
`;

exports[`Player handlers handleUpdateVolumeTo it calls setState and sets player.volume 1`] = `
Object {
  "volume": 1,
}
`;

exports[`Player when ControlsBar is hidden it renders correctly 1`] = `
<div>
  <div
    className="css-3vvba5"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    <video
      height={480}
      onClick={[Function]}
      onLoadedMetadata={[Function]}
      onTimeUpdate={[Function]}
      src={undefined}
    />
  </div>
  <h3 />
</div>
`;

exports[`Player when is in fullscreen mode it renders correctly 1`] = `
<div>
  <div
    className="css-16kk46b"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    <video
      height="100%"
      onClick={[Function]}
      onLoadedMetadata={[Function]}
      onTimeUpdate={[Function]}
      src={undefined}
    />
    <div
      className="css-h8qyac Controls"
    >
      <div
        className="css-yvobmt"
      >
        <input
          className="css-1mwzs46"
          max={0}
          onChange={[Function]}
          size="0.2em"
          step="any"
          type="range"
          value={0}
        />
        <div
          className="css-bg36z2"
        />
      </div>
      <div
        className="css-12alag6"
      >
        <div
          className="css-gijb3s"
        >
          <span
            className="css-bcxn3o PlayAndPause"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M8 5v14l11-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o Sound"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o"
          >
            <div
              className="css-y3p3hm"
            >
              <input
                className="css-qc128d"
                max={1}
                onChange={[Function]}
                size="0.15em"
                step="any"
                type="range"
                value={1}
              />
              <div
                className="css-f8oo8k"
              />
            </div>
          </span>
          <span
            className="css-1g6ztab"
          >
            00:00:00 / 00:00:00
          </span>
        </div>
        <div
          className="css-1dsees0"
        >
          <span
            className="css-bcxn3o Fullscreen"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M5 16h3v3h2v-5H5v2zm3-8H5v2h5V5H8v3zm6 11h2v-3h3v-2h-5v5zm2-11V5h-2v5h5V8h-3z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <h3 />
</div>
`;

exports[`Player when is playing it renders correctly 1`] = `
<div>
  <div
    className="css-drj0bf"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    <video
      height={480}
      onClick={[Function]}
      onLoadedMetadata={[Function]}
      onTimeUpdate={[Function]}
      src={undefined}
    />
    <div
      className="css-h8qyac Controls"
    >
      <div
        className="css-yvobmt"
      >
        <input
          className="css-1mwzs46"
          max={0}
          onChange={[Function]}
          size="0.2em"
          step="any"
          type="range"
          value={0}
        />
        <div
          className="css-bg36z2"
        />
      </div>
      <div
        className="css-12alag6"
      >
        <div
          className="css-gijb3s"
        >
          <span
            className="css-bcxn3o PlayAndPause"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M6 19h4V5H6v14zm8-14v14h4V5h-4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o Sound"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o"
          >
            <div
              className="css-y3p3hm"
            >
              <input
                className="css-qc128d"
                max={1}
                onChange={[Function]}
                size="0.15em"
                step="any"
                type="range"
                value={1}
              />
              <div
                className="css-f8oo8k"
              />
            </div>
          </span>
          <span
            className="css-1g6ztab"
          >
            00:00:00 / 00:00:00
          </span>
        </div>
        <div
          className="css-1dsees0"
        >
          <span
            className="css-bcxn3o Fullscreen"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <h3 />
</div>
`;

exports[`Player with medium it renders correctly 1`] = `
<div>
  <div
    className="css-drj0bf"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
  >
    <video
      height={480}
      onClick={[Function]}
      onLoadedMetadata={[Function]}
      onTimeUpdate={[Function]}
      src={undefined}
    />
    <div
      className="css-h8qyac Controls"
    >
      <div
        className="css-yvobmt"
      >
        <input
          className="css-1mwzs46"
          max={0}
          onChange={[Function]}
          size="0.2em"
          step="any"
          type="range"
          value={0}
        />
        <div
          className="css-bg36z2"
        />
      </div>
      <div
        className="css-12alag6"
      >
        <div
          className="css-gijb3s"
        >
          <span
            className="css-bcxn3o PlayAndPause"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M8 5v14l11-7z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o Sound"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M3 9v6h4l5 5V4L7 9H3zm13.5 3c0-1.77-1.02-3.29-2.5-4.03v8.05c1.48-.73 2.5-2.25 2.5-4.02zM14 3.23v2.06c2.89.86 5 3.54 5 6.71s-2.11 5.85-5 6.71v2.06c4.01-.91 7-4.49 7-8.77s-2.99-7.86-7-8.77z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="css-bcxn3o"
          >
            <div
              className="css-y3p3hm"
            >
              <input
                className="css-qc128d"
                max={1}
                onChange={[Function]}
                size="0.15em"
                step="any"
                type="range"
                value={1}
              />
              <div
                className="css-f8oo8k"
              />
            </div>
          </span>
          <span
            className="css-1g6ztab"
          >
            00:00:00 / 00:00:00
          </span>
        </div>
        <div
          className="css-1dsees0"
        >
          <span
            className="css-bcxn3o Fullscreen"
            onClick={[Function]}
          >
            <svg
              className="css-diyyq9"
              color="hsla(0, 0%, 100%, 0.8)"
              height="1.2em"
              viewBox="0 0 24 24"
              width="1.2em"
            >
              <path
                d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <h3 />
</div>
`;

exports[`Player without props it renders correctly 1`] = `null`;
